src/targets/gpu/mlir.cpp

```diff
@@ -795,7 +795,8 @@ struct mlir_program
     {
         // 1st pipeline to call
         run_high_level_pipeline();
-        if(solution.is_null())
+        std::string tuning_cfg_path = string_value_of(MIGRAPHX_MLIR_TUNING_CFG{});
+        if(solution.is_null() or not tuning_cfg_path.empty())
```

Comment:
If the tuning_cfg_path is set then it will skip setting the solution. Probably should seperate the condition:

```cpp
if(not tuning_cfg_path.empty())
    get_module_tuned();
if(not solution.is_null())
    set_tuning(solution);
```